### DISCUSSION TITLE
Inquiry About Using ShapeID as a Universal Marker for Geoboundaries

### DISCUSSION THREAD

**Discussion by SOORAJTS2001** [wmgeolab/geoBoundaries#4183] [Q&A] [answered]
I am currently working on a project that involves managing geospatial boundaries, and I wanted to inquire if ShapeID could be used as a universally unique marker for identifying these geoboundaries. Its uniqueness and association with spatial data make it seem like a promising candidate for this purpose.

Could you kindly confirm if ShapeID is suitable for this use case or if there are any limitations I should be aware of? Any insights or recommendations you could provide would be greatly appreciated.

Thank you for your time and assistance.

**Comment by DanRunfola** [comment, marked as answer]
Hi Soorajts,

shapeID is tricky - it is based on a hash of the underlying geometry of every file.  So, if the geometry changes in *any way*, so too will the ID.  This results in unique IDs for every geometry, but also results in instability in our IDs across releases if there are even minor changes (i.e., a vertex moves).

We hope - in either 7.0 this year or 8.0 next year - to introduce a new column, likely "staticID", which would be an ID that only changes in response to major changes: i.e., an administrative zone splits or merges with another.  We're working on the underlying algorithms that would let us try to detect that now.  The big advantage of this is it would allow much easier joining of data to geoBoundaries, across different datasets; today, most users are reliant on spatial joins, which inhibits some use cases.
